### PR TITLE
execution: Preserve extensions in parseValue errors

### PR DESCRIPTION
**Pull request by glasser**
Previously, the only fields observed on an error thrown by (for example)
parseValue were `message` and `originalError`. Now, the error itself is
used as the `originalError`; this may be mildly backwards-incompatible
if you added extensions on the error itself which you for some reason
wanted to disappear, but that seems unlikely.

Addresses an issue raised in
https://github.com/apollographql/apollo-server/issues/7178

### CODE REVIEW

**Comment by glasser** [comment]
I'm not sure this is the best solution: another solution would just be to always pass `originalError: error`, and another would be to explicitly copy `extensions: error.extensions` along with `originalError: error.originalError`. (Is there an appropriate pattern for "make a copy of an error but with the message changed"? Perhaps going through `.toJSON()`?)

**Comment by glasser** [comment]
As a workaround, you can throw something like:

```
const e = new GraphQLError('message', {extensions: {my: 'extensions'}});
throw new GraphQLError(e.message, { originalError: e });
```

**Comment by IvanGoncharov** [comment]
> These both also cause my test to pass, and also ensure that extensions from error are honored even if it itself has its own originalError, so in fact originalError: error feels like the best approach. 

@glasser I agree, having correct semantic "chain of errors" make more sense.
Original error for "variable error" is "value error", and original error for "value error".

The entire mechanism of prefixing errors is a workaround for absence of proper way to represent "error chaining" in graphql, see https://github.com/graphql/graphql-spec/issues/893
Moreover, single "variable error" can contain multiple "value error"s and it useful for IDE validation.

TL;DR; I'm for `originalError: error` solution (as breaking change) in `main` and for `originalError: error.originalError ?? error` solution for v15/v16 if needed.

What do you think?

**Comment by glasser** [comment]
@IvanGoncharov OK, I changed this to `originalError: error`. I don't feel too much urgency for a backport (as long as v17 is coming out eventually!) since there's a workaround.

**Comment by yaacovCR** [comment]
Looks like this fix was merged into main as https://github.com/graphql/graphql-js/pull/3837/files

Not sure the exact where’s and whys but thank you @glasser  for working on this!
